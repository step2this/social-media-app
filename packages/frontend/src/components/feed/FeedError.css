.feed-error {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 60px 20px;
  min-height: 300px;
}

.feed-error__message {
  font-size: 16px;
  color: var(--color-error, #d32f2f);
  margin: 0 0 24px 0;
  text-align: center;
}

.feed-error__retry-btn {
  padding: 10px 24px;
  font-size: 14px;
  font-weight: 600;
  color: white;
  background-color: var(--color-primary, #0095f6);
  border: none;
  border-radius: 8px;
  cursor: pointer;
  transition: background-color 0.2s;
}

.feed-error__retry-btn:hover {
  background-color: var(--color-primary-hover, #0081d9);
}
