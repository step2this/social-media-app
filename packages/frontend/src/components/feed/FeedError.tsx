import React from 'react';
import './FeedError.css';

export interface FeedErrorProps {
  message: string;
  onRetry: () => void;
}

/**
 * Error state component for feed
 * Displays error message with retry button
 */
export const FeedError: React.FC<FeedErrorProps> = ({ message, onRetry }) => {
  return (
    <div className="feed-error" role="alert">
      <span className="feed-error__icon" aria-hidden="true">⚠️</span>
      <p className="feed-error__message">{message}</p>
      <button onClick={onRetry} className="feed-error__retry-btn">
        Retry
      </button>
    </div>
  );
};
