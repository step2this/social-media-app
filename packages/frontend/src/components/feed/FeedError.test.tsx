import { describe, it, expect, vi, afterEach } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';
import { FeedError } from './FeedError';

describe('FeedError', () => {
  const mockOnRetry = vi.fn();
  const defaultMessage = 'Something went wrong';

  afterEach(() => {
    vi.clearAllMocks();
  });

  it('renders error message', () => {
    render(<FeedError message={defaultMessage} onRetry={mockOnRetry} />);
    expect(screen.getByText(defaultMessage)).toBeInTheDocument();
  });

  it('displays custom error message prop', () => {
    const customMessage = 'Failed to load feed';
    render(<FeedError message={customMessage} onRetry={mockOnRetry} />);
    expect(screen.getByText(customMessage)).toBeInTheDocument();
  });

  it('shows retry button', () => {
    render(<FeedError message={defaultMessage} onRetry={mockOnRetry} />);
    expect(screen.getByRole('button', { name: /try again/i })).toBeInTheDocument();
  });

  it('calls onRetry when button clicked', () => {
    render(<FeedError message={defaultMessage} onRetry={mockOnRetry} />);
    const button = screen.getByRole('button', { name: /try again/i });
    fireEvent.click(button);
    expect(mockOnRetry).toHaveBeenCalledTimes(1);
  });

  it('has correct CSS classes', () => {
    const { container } = render(<FeedError message={defaultMessage} onRetry={mockOnRetry} />);
    const errorDiv = container.querySelector('.feed-error');
    expect(errorDiv).toBeInTheDocument();
    expect(errorDiv).toHaveClass('feed-error');
  });

  it('matches snapshot', () => {
    const { container } = render(<FeedError message={defaultMessage} onRetry={mockOnRetry} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('has accessible role alert', () => {
    const { container } = render(<FeedError message={defaultMessage} onRetry={mockOnRetry} />);
    const errorDiv = container.querySelector('.feed-error');
    expect(errorDiv).toHaveAttribute('role', 'alert');
  });

  it('retry button has correct text', () => {
    render(<FeedError message={defaultMessage} onRetry={mockOnRetry} />);
    const button = screen.getByRole('button', { name: /try again/i });
    expect(button).toHaveTextContent('Try Again');
  });

  it('error icon displays', () => {
    const { container } = render(<FeedError message={defaultMessage} onRetry={mockOnRetry} />);
    const icon = container.querySelector('.feed-error__icon');
    expect(icon).toBeInTheDocument();
  });

  it('renders without crashing', () => {
    const { container } = render(<FeedError message={defaultMessage} onRetry={mockOnRetry} />);
    expect(container).toBeInTheDocument();
  });
});
