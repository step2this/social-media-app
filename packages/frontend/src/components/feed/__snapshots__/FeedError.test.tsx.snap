// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FeedError > matches snapshot 1`] = `
<div
  class="feed-error"
  role="alert"
>
  <span
    aria-hidden="true"
    class="feed-error__icon"
  >
    ⚠️
  </span>
  <p
    class="feed-error__message"
  >
    Something went wrong
  </p>
  <button
    class="feed-error__retry-btn"
  >
    Try Again
  </button>
</div>
`;
